crates/bitcoin-da/src/helpers/builders.rs

```diff
@@ -87,7 +94,7 @@ fn choose_utxos(
         sum += req_amount;
     }
     if sum >= amount {
-        return Ok((chosen_utxos, sum));
+        return Ok((chosen_utxos, sum, utxos.to_vec()));
```

Comment:
there is a good chance the `required_utxo` is in utxos. pls remove it from utxos.

Reply:
We already do that in:

```
    if let Some(req_utxo) = &prev_utxo {
        // if we don't do this, then we might end up using the required utxo twice
        // which would yield an invalid transaction
        // however using a different txo from the same tx is fine.
        utxos.retain(|utxo| !(utxo.vout == req_utxo.vout && utxo.tx_id == req_utxo.tx_id));
    }
```